src/main/java/Duke.java

```diff
@@ -1,10 +1,129 @@
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        welcomeMessage();
+        Task[] userLists = new Task[] {};
+        Scanner input = new Scanner(System.in);
+        String userInput = input.nextLine();
+
+        //main handler for receiving input
+        while(!userInput.equals("bye")) {
+            //change user input into an array of tokens
+            Tokenise userInputTokens = new Tokenise(userInput);
+            String newUserInput = userInputTokens.removeKeyword();
+            if (userInputTokens.getIsKeyword()) {
+                switch (userInputTokens.getTokens()[0]){
+                case "list":
+                    String allTasks = listTask(userLists);
+                    allTasks = wrapMessage(allTasks);
+                    System.out.println(allTasks);
+                    break;
+                case "todo":
+                    Todo newTodo = new Todo(newUserInput);
+                    userLists = addTask(newTodo, userLists);
+                    break;
+                case "deadline":
+                    //split string into the description and time
+                    String deadlineTime = newUserInput.substring(newUserInput.indexOf("/")+4);
+                    newUserInput = newUserInput.substring(0, newUserInput.indexOf("/"));
+                    Deadline newDeadline = new Deadline(newUserInput.trim(), deadlineTime);
+                    userLists = addTask(newDeadline, userLists);
+                    break;
+                case "event":
+                    //split string into the description and time
+                    String eventTime = newUserInput.substring(newUserInput.indexOf("/")+4);
+                    newUserInput = newUserInput.substring(0, newUserInput.indexOf("/"));
+                    Event newEvent = new Event(newUserInput, eventTime);
+                    userLists = addTask(newEvent, userLists);
+                    break;
+                case "mark":
+                    userLists[Integer.parseInt(userInputTokens.getTokens()[1])].setMark();
+                    System.out.println(
+                            wrapMessage("Nice! I've marked this task as done:\n" +
+                                    userLists[Integer.parseInt(
+                                            userInputTokens.getTokens()[1])].toString()
+                            ));
+                    break;
+                case "unmark":
+                    userLists[Integer.parseInt(userInputTokens.getTokens()[1])].unMark();
+                    System.out.println(
+                            wrapMessage("OK, I've marked this task as not done yet:\n" +
+                                    userLists[Integer.parseInt(
```

Comment:
For this part, userInputTokens.getTokens()[1]) should be assigned to a variable which describes what userInputTokens.getTokens()[1]) represents and the command userLists[Integer.parseInt(userInputTokens.getTokens()[1])].toString() should be broken up into multiple components to improve the code quality as the command is very complex at the moment.